sdk/lib/crt/misc/dbgrpt.cpp

```diff
@@ -62,6 +66,9 @@ struct dbgrpt_char_traits<wchar_t>
 
```

Comment:
Can't these two templates combined together somehow? :thinking:

Reply:
Do you have a suggestion?

Reply:
I tested this on both MSVC and GCC:
```c++
#include <iostream>

template <typename CHAR_T>
struct dbgrpt_char_traits
{
    typedef CHAR_T char_t;

    static const wchar_t* szAssertionMessage;
    static const char_t* szEmptyString;
    static const char_t* szUnknownFile;

    static void OutputDebugString(const char_t* message);
};

typedef dbgrpt_char_traits<char> achar_traits;
typedef dbgrpt_char_traits<wchar_t> wchar_traits;

//
// While MSVC appears to be lenient on this, GCC requires the template<> thing
// at the beginning for ALL static members; see for example:
// https://stackoverflow.com/questions/16493187/specializing-member-sdisplay-requires-template-syntax
// https://stackoverflow.com/questions/12525012/c-specializing-member-requires-template-syntax/12525107
//

template<>
void achar_traits::OutputDebugString(const char* message)
{
    std::cout << "In achar_traits::OutputDebugString, " << message << std::endl;
}

template<>
void wchar_traits::OutputDebugString(const wchar_t* message)
{
    std::wcout << L"In wchar_traits::OutputDebugString, " << message << std::endl;
}

template<> const achar_traits::char_t* achar_traits::szUnknownFile = "ANSI string";
template<> const wchar_traits::char_t* wchar_traits::szUnknownFile = L"UNICODE string";

int main(int argc, char** argv)
{
    achar_traits ach;
    wchar_traits wch;

    std::cout << "Hello World!" << std::endl;
    ach.OutputDebugString("Hi1\n");
    wch.OutputDebugString(L"Hi2\n");

    std::cout << achar_traits::szUnknownFile << std::endl;
    std::wcout << wchar_traits::szUnknownFile << std::endl;

    return 0;
}
```

Reply:
ah yes, thanks

Reply:
By the way, regarding the comment about MSVC vs. GCC (+ the weblinks), I've found it peculiar to see that as soon as you explicitly write the template specializations for the dbgrpt_char_traits structures, as you currently have in the PR, then GCC does not complain anymore about the missing `template<>` in front of the specializations of the static functions.

Reply:
yeah GCC is a mess when it comes to templates.

Reply:
indeed, so I guess I'll keep the struct as-is (saving me from having to litter `template<>` everywhere telling gcc to do it's job.)